# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm
# Commit: 39508d8be196
# Full Hash: 39508d8be19612fbb510707b294bdd730303f6d2
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2023-01-13 21:39:47
# Description:
#   Bug 1809860 - In VideoCaptureAvFoundation account for backends being sharable. r=padenot
#   
#   This is a stop-gap fix to handle camera backends being shared across multiple
#   camera requests. We'll have to follow up with some markers so that the analyzer
#   extension can understand which tracking ids are sharing which backend.
# ==============================================================================

diff -r 32e4b5d65afa -r 39508d8be196 dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm
--- a/dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm	Fri Jan 13 13:54:49 2023 +0000
+++ b/dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm	Fri Jan 13 14:24:59 2023 +0000
@@ -231,6 +231,11 @@
 void VideoCaptureAvFoundation::SetTrackingId(const char* _Nonnull aTrackingId) {
   RTC_DCHECK_RUN_ON(&mChecker);
   MutexLock lock(&api_lock_);
+  if (NS_WARN_IF(mTrackingId.isSome())) {
+    // This capture instance must be shared across multiple camera requests. For now ignore other
+    // requests than the first.
+    return;
+  }
   mTrackingId = Some(nsCString(aTrackingId));
 }
 
